text/0010-bft-signatures.md

```diff
@@ -0,0 +1,291 @@
+The threshold itself can vary depending on the consensus algorithm and on the stage in it that the signatures is produced: Signing a block post consensus requires signatures from only `f+1` orderers (at the expense of latency), while signing during consensus requires signatures from `2f+1` orderers (at the expense of size and computation).
+
+
+# Naive approaches for block replication and block signature verification
+
+### Block replication
+
+In a Byzantine setting, there is a fundamental assumption that there is an upper bound on the number of malicious or failed nodes
+ among all the nodes that run the protocol, and that upper bound is denoted as `f`.
+ In BFT, the ordering service is comprised of `n>3f` for some `f>0`, so we have up to `f` nodes which are either offline, unreachable or malicious,
+ and at least `n-f` nodes that are online, reachable and follow the protocol.
+
+An easy way of ensuring malicious orderers do not refrain from sending newly formed blocks is to simply pull blocks from `f+1` distinct orderers.
+This ensures that even if we happen to connect to all `f` malicious orderers that will not send us blocks, the remaining orderer will, since it is honest.
+
+Obviously, such an approach increases the network bandwidth utilization of the orderers by a factor of `f` which is linear in the number of total ordering service nodes that run the consensus protocol.
+
+### Block signature verification 
+
+A Hyperledger Fabric block is designed to carry an arbitrary number of signatures in its metadata section:
+
+```
+message MetadataSignature {
+    bytes signature_header = 1; // An encoded SignatureHeader
+    bytes signature = 2;       // The signature over the concatenation of the Metadata value bytes, signatureHeader, and block header
+}
+```
+
+The signature header contains the identity of the signer which carries an x509 certificate and the MSP identity that one of its certificate authorities issued that certificate:
+
+```
+message SignatureHeader {
+    // Creator of the message, a marshaled msp.SerializedIdentity
+    bytes creator = 1;
+
+    // Arbitrary number that may only be used once. Can be used to detect replay attacks.
+    bytes nonce = 2;
+}
+```
+
+This means that when signed by `2f+1` (or `f+1`) orderers, each block contains a number of signatures that is linear in the number of ordering service nodes that participate in the consensus protocol.
+When considering an ordering service that can tolerat two failures (`f=2`, hence `n=7`) this means that each block carries  3.5KB of x509 certificates,
+even if the block itself contains only a single transaction (that is often 3.5KB in size itself).
+
+Clearly, encoding identities each time in the signature header of each orderer signature is wasteful and therefore we should strive to eliminate this overhead.
```

Comment:
It is a tradeoff between storage and network bandwidth+verification speed.
Another problem with the static signature collection is that the orderers set in the channel can change, which will make the signature collection difficult to be applicable always.